nh_food_and_fluid/models/food_fluid_review.py

```diff
@@ -176,6 +187,151 @@ def schedule_review(self, spell_activity):
+            return {}
+        fluid_intake = food_fluid_model.\
+            _calculate_total_fluid_intake_from_obs_activities(obs)
+        score = food_fluid_model.calculate_period_score(fluid_intake)
+        user_or_reason = self.get_user_or_reason(review_activity)
+        state = review_activity.state
+        if state == 'cancelled':
+            state = 'Not Completed'
+        return {
+            'score': score,
+            'state': state.title(),
+            'user': user_or_reason
+        }
+
+    def get_review_end(self, period_start, trigger_time):
+        """
+        Get the datetime 1 minute before the next review so we can use this
+        to find observations and reviews that fall within this timeframe
+        :param period_start: Start of the current F&F period
+        :param trigger_time: Time of the review we want the end datetime for
+        :return: Odoo datetime string representation
+        """
+        # find the next trigger time
+        trigger_time_index = self.trigger_times.index(trigger_time)
+        sorted_trigger_times = self.trigger_times[trigger_time_index:] + \
+            self.trigger_times[:trigger_time_index]
+        next_trigger_time = sorted_trigger_times[1]
+        # convert that trigger time to a date
+        next_trigger_datetime = period_start.replace(hour=next_trigger_time)
+        if next_trigger_time < 7:
+            next_trigger_datetime = next_trigger_datetime + timedelta(days=1)
+        # subtract 1 minute
+        next_trigger_datetime = next_trigger_datetime + timedelta(minutes=-1)
+        # return string
+        return next_trigger_datetime.astimezone(pytz.utc).strftime(dtf)
+
+    def get_data_for_trigger_time(
+            self, patient_id, period_start, trigger_time):
+        """
+        Get Observations and Review for
+        :param patient_id: Patient ID
+        :param period_start: Start of the period
+        :param trigger_time: Trigger Time of Review
+        :return: tuple of review activity and obs
+        """
+        user_model = self.env['res.users']
+        review_creator = user_model.search(
+            [
+                ['login', '=', 'food_fluid_review_creator']
+            ]
+        )
+        period_start_datetime = datetime.strptime(period_start, dtf)
+        # get period start for next period
+        period_end_datetime = fields.datetime.context_timestamp(
+            self._cr, review_creator.id, period_start_datetime) \
+            .replace(hour=trigger_time)
+        if trigger_time < 7:
+            day_delta = timedelta(days=1)
+            period_end_datetime = period_end_datetime + day_delta
+        review_time = period_end_datetime.astimezone(pytz.utc).strftime(dtf)
+        review_end = self.get_review_end(period_end_datetime, trigger_time)
+        activity_domain = [
+            ['patient_id', '=', patient_id],
+            ['date_scheduled', '>=', review_time],
+            ['date_scheduled', '<=', review_end],
```

Comment:
Yeah this was just in relation to my other comment above if you decided to change it, otherwise it's redundant 😄